truss/remote/baseten/remote.py

```diff
@@ -78,41 +82,61 @@ def push(  # type: ignore
+                )
+            return dev_version
+
+        # Return the production deployment version.
+        prod_version = get_prod_version_info_from_versions(model_versions)
+        if not prod_version:
+            raise click.UsageError(
+                "No production model found. Run `truss push --publish` then try again."
+            )
+        return prod_version
+
+    @staticmethod
     def _get_service_url_path_and_model_ids(
-        self, model_identifier: ModelIdentifier, published: bool
+        api: BasetenApi, model_identifier: ModelIdentifier, published: bool
     ) -> Tuple[str, str, str]:
+        if isinstance(model_identifier, ModelVersionId):
+            try:
+                model_version = api.get_model_version_by_id(model_identifier.value)
+            except ApiError:
+                raise click.UsageError(
+                    f"Model version {model_identifier.value} not found."
+                )
+            model_version_id = model_version["model_version"]["id"]
+            model_id = model_version["model_version"]["oracle"]["id"]
+            service_url_path = f"/model_versions/{model_version_id}"
+            return service_url_path, model_id, model_version_id
+
         if isinstance(model_identifier, ModelName):
-            model_id, model_versions = get_model_versions_info(
-                self._api, model_identifier.value
+            model_id, model_versions = get_model_versions_info(api, model_identifier)
+            model_version = BasetenRemote._get_matching_version(
+                model_versions, published
             )
-            matching_versions = [
-                model_version
-                for model_version in model_versions
-                # If published is False, then we want to find the draft version
-                if model_version["is_draft"] == (not published)
-            ]
-
-            model_version = matching_versions[0] if matching_versions else None
-            if model_version is None:
-                raise ValueError(
-                    "No appropriate model version found. Run `truss push` then try again."
-                )
             model_version_id = model_version["id"]
             service_url_path = f"/model_versions/{model_version_id}"
-
         elif isinstance(model_identifier, ModelId):
-            model = self._api.get_model_by_id(model_identifier.value)
-            model_id = model["model"]["id"]
-            model_version_id = model["model"]["primary_version"]["id"]
+            model_id, model_versions = get_model_versions_info_by_id(
+                api, model_identifier
+            )
+            # TODO(helen): make this consistent with getting the service via
+            # model_name and respect --published in service_url_path.
+            model_version = BasetenRemote._get_matching_version(
+                model_versions, published
+            )
+            model_version_id = model_version["id"]
```

Comment:
This implementation is now weirdly in between the original implementation and my attempt to unify the `model_id` and `model_name` code paths. It might be more readable to revert this back to the original `model_id` code path and leave these changes for if / when we unify `model_id` and `model_name` paths

Reply:
Yeah, I think for now it might better to revert back for now. It's one less call at least.

Reply:
SG, done